packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -698,3 +708,59 @@ export function isAzureCoreModel(t: SdkType): boolean {
 export function isPagedResultModel(context: TCGCContext, t: SdkType): boolean {
   return context.__pagedResultSet.has(t);
 }
+
+/**
+ * Find corresponding http parameter list for a client initialization or service method parameter
+ * @param method
+ * @param param
+ * @returns
+ */
+export function getHttpOperationParameter(
+  method: SdkServiceMethod<SdkHttpOperation>,
+  param: SdkMethodParameter,
+): (
+  | SdkPathParameter
+  | SdkQueryParameter
+  | SdkHeaderParameter
+  | SdkCookieParameter
+  | SdkBodyParameter
+)[] {
+  const operation = method.operation;
+  const result: (
+    | SdkPathParameter
+    | SdkQueryParameter
+    | SdkHeaderParameter
+    | SdkCookieParameter
+    | SdkBodyParameter
+  )[] = [];
+  const queue: SdkModelPropertyType[] = [param];
+  const visited: Set<SdkModelType> = new Set();
+  while (result.length === 0 && queue.length > 0) {
```

Comment:
I didn't quite get the while condition. In which case it would get into 2nd loop?

Better to add a comment on it.

Reply:
comment added.